CARE-147: Final merge of orchard-upgrade-1.11 plus some additional changes

0liver opened the pull request:
CARE-147

BenedekFarkas committed:
BasicAuthenticationService: Adapting to Orchard 1.11 (prior to release) changes

--HG--
branch : orchard-upgrade-1.11

BenedekFarkas committed:
Extending HqlQueryExtensions and HqlExpressionFactoryExtensions to be able to produce a WhereIdNotIn query

--HG--
branch : orchard-upgrade-1.11

BenedekFarkas committed:
Merge branch 'dev' into orchard-upgrade-1.11

--HG--
branch : orchard-upgrade-1.11

Piedone committed:
Merge branch 'dev' into orchard-upgrade-1.11

--HG--
branch : orchard-upgrade-1.11

Piedone committed:
Code styling

--HG--
branch : orchard-upgrade-1.11

BenedekFarkas committed:
Merge branch 'dev' into orchard-upgrade-1.11

--HG--
branch : orchard-upgrade-1.11

BenedekFarkas committed:
Merge branch 'dev' into orchard-upgrade-1.11

--HG--
branch : orchard-upgrade-1.11

BenedekFarkas committed:
Code cleanliness fix (breaking change) in HqlExpressionFactoryExtensions

--HG--
branch : orchard-upgrade-1.11

BenedekFarkas committed:
Merge branch 'dev' into orchard-upgrade-1.11

--HG--
branch : dev

BenedekFarkas committed:
ScheduledTaskManagerExtensions: Code styling and adding RescheduleTaskOrCreate extension method

BenedekFarkas committed:
Moving HqlQueryExtensions and HqlExpressionFactoryExtensions to Helpful Extensions

0liver committed:
remove obsolete and unused IShapeOutputGenerator

BenedekFarkas committed:
Merge pull request #28 from Lombiq/issue/COLI-263

COLI-263: remove obsolete and unused IShapeOutputGenerator

BenedekFarkas committed:
Removing unused reference to Orchard.Core from the web.config as there's no project reference

BenedekFarkas committed:
Making HelpfulLibraries compatible with static Razor compilation

BenedekFarkas committed:
Restricting C#/Razor support to version 7.3

BenedekFarkas committed:
Code styling in RequireBasicAuthorizationAttribute and fixing that Disabled user accounts should not be authorized

Commit:
Compiling project in CustomForms solution

Commit:
Refactoring FileCleanserHelper

 - better exception throwing
 - more compact code
 - improved docs

Commit:
Adding StringExtensions with If, Join, and Split

Commit:
Adding file to project, add Feature attribute and summary

Commit:
Split examples into syntax-highlighted blocks

Commit:
Add StringExtensions examples

Commit:
Replace unpredictable TABs with spaces

Commit:
Beautiful code blocks for all!

Commit:
Convert HL to a Web Project to make it publishable

Commit:
Minor changes

Commit:
Renaming publishing profile 1/2

Commit:
Renaming publishing profile 2/2

Commit:
Merge remote-tracking branch 'origin/orchard-upgrade-1.11' into dev-orchard-1

Commit:
Improve documentation

0liver committed:
Merge pull request #100 from Lombiq/issue/CARE-147-merge

CARE-147: Merge `orchard-upgrade-1.11` into `dev-orchard-1`

Commit:
Revert "Merge pull request #100 from Lombiq/issue/CARE-147-merge"

This reverts commit 878c635752b70cd7ac1240f673f73dc9610a16e6, reversing
changes made to 907924f89e4d6eb2ef9062e7034869b8549d4da6.

Commit:
Merge remote-tracking branch 'origin/issue/CARE-147-merge' into issue/CARE-147

# Conflicts:
#	Piedone.HelpfulLibraries.csproj

Commit:
Remove obsolete documentation

Commit:
Re-update <TargetFrameworkVersion> to 4.6.1 to make compilation work

Errors with 4.5.2:
~~~~~~~~~~~
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2203,5): warning MSB3274: The primary reference "Orchard.Alias, Version=1.10.1.0, Culture=neutral, processorArchitecture=MSIL" could not be resolved because it was built against the ".NETFramework,Version=v4.6.1" framework. This is a higher version than the currently targeted framework ".NETFramework,Version=v4.5.2".

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2203,5): warning MSB3274: The primary reference "Orchard.Caching, Version=1.10.1.0, Culture=neutral, processorArchitecture=MSIL" could not be resolved because it was built against the ".NETFramework,Version=v4.6.1" framework. This is a higher version than the currently targeted framework ".NETFramework,Version=v4.5.2".

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2203,5): warning MSB3274: The primary reference "Orchard.Framework, Version=1.10.1.0, Culture=neutral, processorArchitecture=MSIL" could not be resolved because it was built against the ".NETFramework,Version=v4.6.1" framework. This is a higher version than the currently targeted framework ".NETFramework,Version=v4.5.2".

Commit:
Fix missing Orchard.Users.dll

Commit:
Revert breaking change in an Orchard version >1.10

Commit:
Merge branch 'issue/CARE-147' into dev-orchard-1-care

# Conflicts:
#	Libraries/Authentication/BasicAuthenticationService.cs
#	Piedone.HelpfulLibraries.csproj

Commit:
Align reference for System.Web.WebPages.dll with others above

Commit:
Re-add <LangVersion>7.3</LangVersion>

Commit:
Restore the changes from orchard-upgrade-1.11 after a bad merge in #7cd15b26

Commit:
Merge branch 'dev-orchard-1-care' into dev-orchard-1

... without the CARE-specific changes

BenedekFarkas commented on Libraries/Utilities/HqlQueryExtensions.cs:
```diff
@@ -1,25 +0,0 @@
-            query.Where(a => a.ContentItem(), p => p.PartitionedExpression((e, i) => e.In("Id", i), ids, partitionSize));
-        }
-    }
-}
```
Was this moved to Lombiq.Projections?

0liver replied:
I don't know, but those removals are from the `orchard-upgrade-1.11` branch.

BenedekFarkas replied:
My bad, I moved it to HelpfulExtensions some time ago.

BenedekFarkas commented on Libraries/Utilities/HqlExpressionFactoryExtensions.cs:
```diff
@@ -1,68 +0,0 @@
-            expressionFactory.Or(lhs => partitionExpression(lhs, values.Take(partitionSize).ToArray()), rhs => BuildPartitionExpression(rhs, partitionExpression, values.Skip(partitionSize), partitionSize));
-        }
-    }
-}
```
Was this moved to Lombiq.Projections?

0liver replied:
I don't know, but those removals are from the `orchard-upgrade-1.11` branch.

BenedekFarkas replied:
My bad, I moved it to HelpfulExtensions some time ago.

BenedekFarkas commented:
@I3undy @domonkosgabor please check your project (you know which one! :D), FYI this is an open-source repo. ;)

BenedekFarkas commented on Libraries/Utilities/StringExtensions.cs:
```diff
@@ -0,0 +1,63 @@
+    /// This class provides useful methods for string usage and manipulation, usually safe to use with empty and <see
+    /// langword="null"/> strings.
+    /// </summary>
+    [OrchardFeature("Piedone.HelpfulLibraries.Utilities")]
```
```suggestion

```
OrchardFeature attribute is useless on a static class.

0liver replied:
That makes sense 😀

0liver replied:
But still: `AsyncHelper`, `ContainerBuilderExtensions`, and probably other `static` classes in the _Utilities_ folder **do** carry that attribute. So - what gives?

BenedekFarkas replied:
`OrchardFeature` only applies/matters to stuff that descends from `IDependency`, so it's just an oversight.

0liver replied:
I'll remove it from all the static classes in Utilities then, OK?

0liver replied:
Hm, you were fast! What now?

BenedekFarkas replied:
> I'll remove it from all the static classes in Utilities then, OK?

Allrighty!

Commit:
Revert TargetFramework to 4.5.2

Commit:
Update module version to 1.10 due to the batch of changes.

Commit:
Update module version to 1.10 due to the batch of changes.

Commit:
Remove [OrchardFeature("Piedone.HelpfulLibraries.Utilities")] from static classes

- because it does not have any effect there

Commit:
Remove unused using stmts and duplicate empty lines, format comments and code

Commit:
Refactor and improve MimeAssistant:

 - add parameter name to exception
 - use .Substring() instead of .Remove()
 - avoid duplicate dictionary lookup

Commit:
Merge branch 'dev-orchard-1-care' into issue/CARE-147-final-merge

BenedekFarkas commented on Libraries/Utilities/FreezableBase.cs:
```diff
@@ -1,10 +1,10 @@
+    /// Base for classes implementing the interface <see cref="IFreezable"/>.
     /// </summary>
     /// <remarks>
     /// This is a lightweight appraoch to the Freezable class from WPF.
```
```suggestion
    /// This is a lightweight approach to the Freezable class from WPF.
```

BenedekFarkas committed:
Fixing typo in Libraries/Utilities/FreezableBase.cs
